### ISSUE TITLE
Difference between detaching an project created using expo and react-native init command

### ISSUE THREAD

**Issue by kaulsalil88** [expo/expo#1412]
I know this is a slightly vague question but has me puzzled for the last few days . I was trying to  use realm in my project but got to know that expo does not support realm . But i still implemented the same because my project is in the detached state  . I assumed that expo is basically a super set of the react-native code base  . Post implementing it it worked properly on the android project  but didnt work on the ios project .  Realm claim that they support react-native . So what is causing it to break in an detached project . Some pointers would be really help full and enable one to make better decision about which library to use with a detached expo project . P.S.: I know i should have created a project using react-native init command and integrate realm and report my findings , will try to do the same ASAP .

**Comment by AdamJNavarro** [comment]
The main differences between a fresh projected created by react-native init and detaching an Expo project is the version of react-native that will be used -- with Expo we use our own fork of RN and tend to be a version or two behind due to our release cycle while using the react-native CLI you will be using the current, stable version of RN. Also, detaching will automatically install all the dependencies (ExpoKit and such) needed to let you use APIs from the Expo SDK from a detached project.

**Comment by kaulsalil88** [comment]
@AdamJNavarro  thanks for the reply .  Still i  am not able to figure out why a library like realm needs to be integrated into the expo sdk  for it to work in the detached form .

**Comment by jesseruder** [comment]
The Realm React Native library includes some Java and Obj-C code that has to be compiled into the app for the JS side of the library to work. Realm does support React Native but Expo doesn't include all of the Realm Java and Obj-C code needed to make Realm work (neither does a raw React Native project!), so you have to run `react-native link` to add the extra code to your project. I'd recommend reading this section of the docs: https://docs.expo.io/versions/latest/introduction/faq.html#how-do-i-add-custom-native-code-to-my-expo-project

**Comment by kaulsalil88** [comment]
@jesseruder thanks for clarifying things . Please close the issue .

**Comment by LetsNurtureGit** [comment]
Hello.. I have a question that If i made a project through expo start and then at the middle I require some codes to be run on android and iOS folders. Then what to do ? And after that will the expo modules work again? And after changing into the iOS and android I again want to go to expo and start development again. Then what to do?

**Comment by LetsNurtureGit** [comment]
Hello.. I have a question that If i made a project through expo start and then at the middle I require some codes to be run on android and iOS folders. Then what to do ? And after that will the expo modules work again? And after changing into the iOS and android I again want to go to expo and start development again. Then what to do?

Also If I eject using Expo eject and after that do I have to write the native codes in android and iOS exclusively or writing the codes again in JS work? 
And after ejecting the project and adding something to the JS code again then how will that save to the expo project.
And if I had changed something in the android and iOS folders will that reflect in the main project too?

**Comment by LetsNurtureGit** [comment]
Hello. I have a question that If I made a project through expo start and then at the middle I require some codes to be run on Android and iOS folders. Then what to do? And after that will, the expo modules work again? And after changing into the iOS and Android I again want to go to the expo and start development again. Then what to do?

Also If I eject using Expo eject and after that do I have to write the native codes in Android and iOS exclusively or writing the codes again in JS work? 
And after ejecting the project and adding something to the JS code again then how will that save to the expo project.
And if I had changed something in the Android and iOS folders will that reflect in the main project too?
